src/util/cmdline_definition.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************\
+
+Module: Command line parsing
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+// \file Command line options definition
+
+#ifndef CPROVER_UTIL_CMDLINE_DEFINITION_H
+#define CPROVER_UTIL_CMDLINE_DEFINITION_H
+
+#include <list>
+#include <string>
+#include <vector>
+
+#include "optional.h"
+
+class json_arrayt;
+class xmlt;
+
+struct cmdline_optiont
+{
+  struct argumentt
+  {
+    argumentt(std::string name, std::string type);
+
+    std::string name;
+    std::string type;
+  };
+
+  typedef std::string deprecatedt;
+
+  cmdline_optiont(
+    std::string name,
+    optionalt<argumentt> argument,
+    std::string help_text,
+    std::string option_group,
+    optionalt<deprecatedt> deprecated);
```

Comment:
Wouldn't it make for a very easy (and IMHO) nice extension to have a second constructor that takes a `std::function` for argument parsing (possibly achieving what @LAJW suggested)?

Reply:
That requires a bit more thinking as many options are not independent... I have a few ideas...

Reply:
Speaking of dependent options: it would actually be nice if one had a way to specify implied options (e.g., `--z3` implies `--smt2`). But I suppose the wish-list is unbounded (what about conflicting options...) 😇